app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/CourseReviewInfoFragment.kt

```diff
@@ -0,0 +1,23 @@
+package com.github.sdp.ratemyepfl.fragment.review
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.View
+import android.widget.TextView
+import androidx.fragment.app.activityViewModels
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.viewmodel.ReviewViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class CourseReviewInfoFragment : Fragment(R.layout.fragment_course_review_info) {
+
+    private val viewModel by activityViewModels<ReviewViewModel>()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.getReviewable().observe(viewLifecycleOwner){
+            view.findViewById<TextView>(R.id.id_course_info).text = it!!.id
```

Comment:
Here is a typical example of such practices. You have to cast a !! on something that is supposed (in the known state of the app) to be non-null. If it was non-null beforehand, we wouldn't need this at all.

Reply:
That has nothing to do with what you said in earlier comment. The nullable here comes from the getExtra method for Intents.

Reply:
I don't think we can get rid of this one. A repo function returns a null value, if nothing is found in the DB.